### DISCUSSION TITLE
What is the eigenvector representation?

### DISCUSSION THREAD

**Discussion by thefacemyer** [QuSpin/QuSpin#565] [Q&A] [answered]
I'm doing exact diagonalization, but I'm confused as to how the bases are represented. When I print the spin_basis_1d basis, I get three columns (array/fock/integer). Which representation is used to call a specific fock state? The array index or the integer representation? Lastly, when I print the eigenvector, say V[0], it gives me an array of cofactors. How are these eigenvector array indices matched with the fock states? E.g. is the zeroth array element of the eigenvector corresponding to the zeroth fock state using the first column (i.e. the array index) or the third column (i.e. the integer representation)?

**Comment by thefacemyer** [comment, marked as answer]
The basis is the fock-state basis. The issue was that I only printed single rows of the cumulative eigenspace, not each individual eigenstate (i.e. column of the space). This was giving me very strange results, which is why I posed the question.

**Comment by mgbukov** [comment]
Hi, take a look at this (and related) example (s), I think it contains the
answers to your questions:

https://weinbe58.github.io/QuSpin/examples/example00.html#example-00-label

Marin

**Reply by hz-xiaxz** [reply]
Hi I think the link has lost, could you repost it? I'm also curious about this problem.

**Reply by hz-xiaxz** [reply]
oh sorry I think it's here https://quspin.github.io/QuSpin/examples/example00.html#example-00-label
